lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -302,4 +288,44 @@ public function matching(Criteria $criteria)
 
         return new LazyCriteriaCollection($persister, $criteria);
     }
+
+    /**
+     * Resolves a magic method call to the proper existent method at `EntityRepository`.
+     *
+     * @param string $method    The method to call
+     * @param string $by        The property name used as condition
+     * @param array  $arguments The arguments to pass at method call
+     *
+     * @throws ORMException If the method called is invalid.
+     *
+     * @return mixed
+     */
+    private function resolveMagicCall($method, $by, array $arguments = [])
+    {
+        $argsCount = count($arguments);
+
+        if (0 === $argsCount) {
+            throw ORMException::findByRequiresParameter($method . $by);
+        }
+
+        $fieldName = lcfirst(\Doctrine\Common\Util\Inflector::classify($by));
+
+        if ($this->_class->hasField($fieldName) || $this->_class->hasAssociation($fieldName)) {
```

Comment:
Should throw an exception when this is not true: move the exception throwing up :-)